serialize/Deserializer.cpp

```diff
@@ -1,13 +1,99 @@
 // Copyright (c) 2014-2017 Josh Blum
+//                    2020 Nicholas Corgan
 // SPDX-License-Identifier: BSL-1.0
 
 #include "SerializeCommon.hpp"
 #include <Pothos/Framework.hpp>
 #include <Poco/ByteOrder.h>
 #include <Poco/Format.h>
+
 #include <sstream>
 #include <cstring>
 #include <cassert>
+#include <numeric>
+
+/*
+ * Utility code
+ */
+
+static inline size_t getTotalBufferSize(const std::vector<Pothos::BufferChunk>& buffers)
+{
+    return std::accumulate(
+               buffers.begin(),
+               buffers.end(),
+               0ULL,
+               [](size_t num, const Pothos::BufferChunk& buffer)
+               {
+                   return (num + buffer.length);
+               });
+}
+
+static void clearBytesFromBuffers(
+    std::vector<Pothos::BufferChunk> &buffers,
+    const size_t numBytes)
+{
+    assert(!buffers.empty());
+
+    const auto totalBufferSize = getTotalBufferSize(buffers);
+    assert(numBytes <= totalBufferSize);
+
+    if(numBytes == 0) return;
+    else if(numBytes == totalBufferSize) buffers.clear();
+    else
+    {
+        auto bytesRemaining = numBytes;
+
+        while(bytesRemaining > 0)
+        {
+            assert(!buffers.empty());
+
+            if(bytesRemaining > buffers[0].length)
+            {
+                bytesRemaining -= buffers[0].length;
+                buffers[0].length = 0;
+            }
+            else
+            {
+                buffers[0].address += bytesRemaining;
+                buffers[0].length -= bytesRemaining;
+                bytesRemaining = 0;
+            }
```

Comment:
given you have to do the compare anyway, it might be better to compute `bytesToRemove = min(bytesRemaining, buffers[0].length)`

and just have the one statement
```
                buffers[0].address += bytesToRemove;
                buffers[0].length -= bytesToRemove;
                bytesRemaining -= bytesToRemove;
                ```